packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -528,25 +463,19 @@ internal MethodProvider BuildPersistableModelGetFormatFromOptionsObjectDeclarati
         }
 
         /// <summary>
-        /// Builds the serialization constructor for the model.
+        /// Finds the serialization constructor for the model.
         /// </summary>
-        /// <returns>The constructed serialization constructor.</returns>
-        internal ConstructorProvider BuildSerializationConstructor()
+        /// <returns>The serialization constructor.</returns>
+        private ConstructorProvider? GetSerializationConstructor()
         {
-            var (serializationCtorParameters, serializationCtorInitializer) = BuildSerializationConstructorParameters();
+            if (_model.Constructors.Count == 1)
+            {
+                return _model.Constructors[0];
+            }
 
-            return new ConstructorProvider(
-                signature: new ConstructorSignature(
-                    Type,
-                    $"Initializes a new instance of {Type:C}",
-                    MethodSignatureModifiers.Internal,
-                    serializationCtorParameters,
-                    Initializer: serializationCtorInitializer),
-                bodyStatements: new MethodBodyStatement[]
-                {
-                    GetPropertyInitializers()
-                },
-                this);
+            return _model.Constructors.FirstOrDefault(ctor =>
```

Comment:
I think the ModelProvider should have a public property `FullConstructor` vs looking based on this criteria.